src/modules/cluster/cluster-detail/cluster-detail.js

```diff
@@ -9,19 +9,28 @@
     function clusterDetailController($state, $stateParams, utils) {
 
         var vm = this;
+        vm.tabList = {};
+        vm.tabName = "";
         vm.setTab = setTab;
         vm.isTabSet = isTabSet;
 
         vm.clusterId = $stateParams.clusterId;
-        vm.tabList = {
-            Volumes: 1
-        };
-        vm.activeTab = vm.tabList.Volumes;
 
         utils.getObjectList("Cluster").then(function(list) {
             vm.cluster = list.filter(function(cluster) {
                 return cluster.cluster_id === vm.clusterId;
             })[0];
+
+            /* Setting up the tab based on ceph/gluster */
+            if(vm.cluster.sds_name === "gluster") {
```

Comment:
Why are we checking condition here, the tabs should be generated based on the cluster context.